### DISCUSSION TITLE
Can input be avoided with "--update-all"?

### DISCUSSION THREAD

**Discussion by boidolr** [ast-grep/ast-grep#906] [Q&A] [answered]
When using `sg scan --update-all` the behaviour for warn/info/hint changes: instead of outputting them, they now want to have input. As a consequence, its not possible to run `sg scan --update-all` in a process without input (as it will crash).

Is there a way to avoid/disable the interactive print in this situation?

**Comment by HerringtonDarkholme** [comment]
Hi, it is possible to make update-all compatible with --stdin, as tracked in https://github.com/ast-grep/ast-grep/issues/655 

That said, would you like to tell your use case so I can understand better? Looks like you are using `sg --stdin` with `--update-all`?

The command ` sg -p 'test($A)' -l js --stdin -r 'mock($A)'`

now produces the output

```
test(123)
STDIN
@@ -0,1 +0,1 @@
1  │-test(123)
  1│+mock(123)
```

I don't know if it meets your use case. But it is not machine-readable output.

**Comment by boidolr** [comment]
What I am trying to do is to run`ast-grep` as a `pre-commit` hook. As such, the package is installed from the npm package ("@ast-grep/cli") because it reliably can install and run for different operating systems. `ast-grep` is then run via the `bin` link.

Using `sg scan --update-all` (without providing other options) if there are rules that have no fix, and only warn, I end up with:
> thread 'main' panicked at crates/cli/src/print/interactive_print.rs:52:50

For an example setup see https://github.com/boidolr/pre-commit-ast-grep
(btw: if you prefer to provide a hook in your organization, I am happy to remove that repo again)

**Reply by HerringtonDarkholme** [reply]
Thanks! I got your point. So for the following command, for example,

```
echo | sg scan --update-all
```

will crash at the moment. right? I will give it a look later. Thanks for the report.

-----
>  (btw: if you prefer to provide a hook in your organization, I am happy to remove that repo again)

I am more than happy if you want to contribute! Thanks!

**Comment by HerringtonDarkholme** [comment]
Fixed in 0ad1b6087d0d6da05bc48561d531cfaf802a3ce7

**Reply by HerringtonDarkholme** [reply, marked as answer]
also shipped in 0.19.1
